internal/client/client_interceptor.go

```diff
@@ -34,8 +38,38 @@ func NewInterceptedClient(client Client) Client {
 }
 
 func (r interceptedClient) SendRequest(ctx context.Context, addr string, req *tikvrpc.Request, timeout time.Duration) (*tikvrpc.Response, error) {
+	var tenantRPCInterceptor interceptor.RPCInterceptor
+	if TenantKVControllor != nil {
+		tenantRPCInterceptor = func(next interceptor.RPCInterceptorFunc) interceptor.RPCInterceptorFunc {
+			return func(target string, req *tikvrpc.Request) (*tikvrpc.Response, error) {
+				reqInfo := resourcegroup.MakeRequestInfo(req)
+				err := TenantKVControllor.OnRequestWait(ctx, "demo", reqInfo)
```

Comment:
we can get the  resource name from Request after https://github.com/tikv/client-go/pull/650.

Reply:
Got it. I will update this PR ASAP after https://github.com/tikv/pd/pull/5811.